src/theory/quantifiers/ceg_instantiator.h

```diff
@@ -138,12 +143,125 @@ class SolvedForm {
   }
 };
 
+/** instantiation effort levels */
+enum InstEffort
+{
+  // uninitialized
+  INST_EFFORT_NONE,
+  // standard effort level
+  INST_EFFORT_STANDARD,
+  // standard effort level, but we have used model values
+  INST_EFFORT_STANDARD_MV,
+  // full effort level
+  INST_EFFORT_FULL
+};
+
 /** Ceg instantiator
  *
  * This class manages counterexample-guided quantifier instantiation
  * for a single quantified formula.
+ *
+ * For details on counterexample-guided quantifier instantiation
+ * (for linear arithmetic), see Reynolds/King/Kuncak FMSD 2017.
  */
 class CegInstantiator {
+ public:
+  CegInstantiator(QuantifiersEngine* qe,
+                  CegqiOutput* out,
+                  bool use_vts_delta = true,
+                  bool use_vts_inf = true);
+  virtual ~CegInstantiator();
+  /** check
+   * This adds instantiations based on the state of d_vars in current context
+   * on the output channel d_out of this class.
+   */
+  bool check();
+  /** presolve for quantified formula
+   *
+   * This initializes formulas that help static learning of the quantifier-free
+   * solver. It is only called if the option --cbqi-prereg-inst is used.
+   */
+  void presolve(Node q);
+  /** Register the counterexample lemma
+   *
+   * lems contains the conjuncts of the counterexample lemma of the quantified
+   *   formula we are processing. The counterexample lemma is the formula
+   *   { ~phi[e/x] } in Figure 1 of Reynolds et al. FMSD 2017.
+   * ce_vars contains the variables e. Notice these are variables of
+   *   INST_CONSTANT kind, since we do not permit bound variables in
+   *   assertions.
```

Comment:
Is the indent here on purpose?

Reply:
it was although perhaps this looks strange in this case, I'll fix.